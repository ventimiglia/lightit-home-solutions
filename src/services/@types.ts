export type ApiResponse = {
  name: string;
  items: Item[];
};

export type Item = {
  name: string;
  img: string;
};
