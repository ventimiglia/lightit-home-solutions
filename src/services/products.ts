import { ApiResponse } from "./@types";

export const getProductsByCategory = async (category: string | null) => {
  try {
    const response = await fetch(`https://us-central1-prueba-front-280718.cloudfunctions.net/${category}`);
    if (!response.ok) {
      throw new Error("Error al obtener los productos por categoría");
    }
    const data = await response.json();
    return data as ApiResponse[];
  } catch (error) {
    console.error("Ocurrió un error al obtener los productos");
    throw error;
  }
};
