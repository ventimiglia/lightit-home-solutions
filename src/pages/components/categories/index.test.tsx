import { render, fireEvent } from "@testing-library/react";
import Categories from "./";
import { ApiResponse } from "../../../services/@types";

const mockApiResponse: ApiResponse = {
  name: "Category 1",
  items: [
    {
      name: "Product 1",
      img: "image.jpg",
    },
  ],
};

describe("Categories", () => {
  it("renders the categories component", () => {
    const { container, getByText } = render(
      <Categories>
        <Categories.Title>Title</Categories.Title>
        <Categories.Category item={mockApiResponse} onClick={() => {}} />
      </Categories>
    );

    const titleElement = getByText("Title");
    expect(titleElement).toBeInTheDocument();

    const categoryNameElement = getByText(mockApiResponse.name);
    expect(categoryNameElement).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("calls onClick event handlers when Category is selected", () => {
    const categoryHandler = jest.fn();

    const { getByText } = render(
      <Categories>
        <Categories.Category item={mockApiResponse} onClick={categoryHandler} />
      </Categories>
    );

    const category = getByText(mockApiResponse.name);
    fireEvent.click(category);

    expect(categoryHandler).toHaveBeenCalledTimes(1);
  });
});
