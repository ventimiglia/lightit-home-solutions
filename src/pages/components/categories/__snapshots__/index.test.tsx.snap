// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories renders the categories component 1`] = `
<div>
  <article
    class="flex flex-col gap-4 w-full"
  >
    <h2
      class="text-2xl font-bold capitalize text-left"
    >
      Title
    </h2>
    <button
      class="flex items-center justify-between p-3 bg-white rounded-md w-full transition duration-300 ease-in-out hover:bg-primary hover:text-white transform hover:scale-105 focus:outline-none focus:ring-2 focus:ring-primary"
    >
      <span>
        Category 1
      </span>
      <img
        class="rotate-180"
        src="chevron.png"
        width="15"
      />
    </button>
  </article>
</div>
`;
